mbbsd/bbs.c

```diff
@@ -433,6 +433,43 @@ int IsFreeBoardName(const char *brdname)
     return 0;
 }
 
+#ifdef POST_COMMENT_WITH_TAIWAN_PHONE_ONLY
+/*
+ * Check if user phone address is registered and from Taiwan
+ *
+ * This function will check is current user have already register their
+ * phone number with verifydb library.
+ * The verify process is modified from verifydb_check_vmethod_unused,
+ * it will pass cuser.userid and cuser.firstlogin as parameter and find
+ * VMETHOD_SMS record.
+ *
+ * Note that, ptt system might has two different user has the same userid
+ * but won't be happen in same time, so we need firstlogin as second
+ * parameter, and ptt accept Taiwan phone register only since Jul. 2021
+ * so we can just verify whether in database.
+ * If someday anyone want to check country code, please see ValidatePhone
+ * But please note that some country is shared country code eg. US and
+ * Canada is shared +1 and Taiwanese +886 is unofficially allocated.
+ *
+ * Return true when user has verified phone, false when no register record
+ */
```

Comment:
This is badly worded. Our system simply implements some form of SMS verification. The fact that only monthly plan holders in Taiwan can pass is a restriction of the backend system we use, not our code.

And nor are you actually checking the phone number here. So this should simply be named something like `is_user_sms_verified()`.